jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceCollection.java

```diff
@@ -130,6 +125,15 @@ public Resource resolve(String subUriPath)
         Resource addedResource = null;
         for (Resource res : _resources)
         {
+            // since we allow non-directory resources, we need a way to resolve them
+            // -> match if the subUriPath contains no / and matches the resource's filename.
+            // TODO review this
+            String subUriStringNotPrependedBySlash = subUriPath;
+            if (subUriStringNotPrependedBySlash.startsWith("/"))
+                subUriStringNotPrependedBySlash = subUriStringNotPrependedBySlash.substring(1);
+            if (!res.isDirectory() && res.exists() && !subUriStringNotPrependedBySlash.contains("/") && res.getPath().getFileName().toString().equals(subUriStringNotPrependedBySlash))
+                return res;
```

Comment:
I don't understand this code?  resolving against a non directory should just result in null.  Perhaps we just skip non directory resources?

Reply:
The idea is that if you as a ResourceCollection to resolve `abc.txt` and it happens to contain a non-directory resource named `abc.txt` then it's going to serve that resource.

I'm not saying this mechanism is a good idea, but if we want to allow ResourceCollection to contain non-directory resources, we must have some form of resolving logic that can match the non-dir resources.

Reply:
Why do we want `ResourceCollection` to contain non-directory resources?    What is the use-case?